sesman/session.c

```diff
@@ -374,7 +374,9 @@ session_get_avail_display_from_chain(void)
 
     display = g_cfg->sess.x11_display_offset;
 
-    while ((display - g_cfg->sess.x11_display_offset) <= g_cfg->sess.max_sessions)
+    /* positive max_sessions value limits the max number of sessions */
+    while (g_cfg->sess.max_sessions <= 0 ||
+           (display - g_cfg->sess.x11_display_offset) <= g_cfg->sess.max_sessions)
```

Comment:
Ideally, we should not even get there if the session limit is exhausted. Moreover, some displays may be unavailable for unrelated reasons, not because they are used by existing xrdp sessions.